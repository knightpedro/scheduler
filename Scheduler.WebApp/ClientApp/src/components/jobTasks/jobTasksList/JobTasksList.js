import React, { useMemo } from "react";
import moment from "moment";
import { Link, generatePath } from "react-router-dom";
import Routes from "../../../routes";
import { BasicTable, SearchColumnFilter } from "../../common/tables";

const TIME_FORMAT = "HH:mm DD/MM/YYYY";

const JobTasksList = ({ jobTasks }) => {
    const createPath = id => generatePath(Routes.jobTasks.DETAIL, { id });

    const columns = useMemo(
        () => [
            {
                Header: "Description",
                accessor: "description",
                Cell: ({ row }) => (
                    <Link to={createPath(row.original.id)}>
                        {row.values.description}
                    </Link>
                ),
                Filter: SearchColumnFilter,
            },
            {
                Header: "Start",
                accessor: "start",
                className: "text-center",
                Cell: ({ cell: { value } }) => value.format(TIME_FORMAT),
                disableFilters: true,
                sortType: "momentSort",
            },
            {
                Header: "End",
                accessor: "end",
                className: "text-center",
                Cell: ({ cell: { value } }) => value.format(TIME_FORMAT),
                disableFilters: true,
                sortType: "momentSort",
            },
        ],
        []
    );

    const data = useMemo(
        () =>
            jobTasks.map(task => ({
                id: task.id,
                description: task.description,
                start: moment(task.start),
                end: moment(task.end),
            })),
        [jobTasks]
    );

    const initialTableState = useMemo(
        () => ({
            sortBy: [
                {
                    id: "start",
                    desc: true,
                },
            ],
        }),
        []
    );

    return (
        <BasicTable
            columns={columns}
            data={data}
            initialState={initialTableState}
        />
    );
};

export default JobTasksList;
