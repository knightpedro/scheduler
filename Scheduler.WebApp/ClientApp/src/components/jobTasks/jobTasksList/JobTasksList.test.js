import React from "react";
import { shallow } from "enzyme";
import JobTasksList from "./JobTasksList";

const jobTasksStub = [
    {
        id: 1,
        description: "Test Task 1",
        start: "2020-02-29T19:00:00.0000000Z",
        end: "2020-02-29T21:00:00.0000000Z",
    },
    {
        id: 2,
        description: "Test Task 2",
        start: "2020-02-28T19:00:00.0000000Z",
        end: "2020-02-28T21:00:00.0000000Z",
    },
];

describe("JobTasksList", () => {
    it("renders correctly", () => {
        const wrapper = shallow(<JobTasksList jobTasks={jobTasksStub} />);
        expect(wrapper).toMatchSnapshot();
    });
});
