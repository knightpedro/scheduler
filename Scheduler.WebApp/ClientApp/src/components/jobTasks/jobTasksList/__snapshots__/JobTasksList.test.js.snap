// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobTasksList renders correctly 1`] = `
<BasicTable
  columns={
    Array [
      Object {
        "Cell": [Function],
        "Filter": [Function],
        "Header": "Description",
        "accessor": "description",
      },
      Object {
        "Cell": [Function],
        "Header": "Start",
        "accessor": "start",
        "className": "text-center",
        "disableFilters": true,
        "sortType": "momentSort",
      },
      Object {
        "Cell": [Function],
        "Header": "End",
        "accessor": "end",
        "className": "text-center",
        "disableFilters": true,
        "sortType": "momentSort",
      },
    ]
  }
  data={
    Array [
      Object {
        "description": "Test Task 1",
        "end": "2020-02-29T21:00:00.000Z",
        "id": 1,
        "start": "2020-02-29T19:00:00.000Z",
      },
      Object {
        "description": "Test Task 2",
        "end": "2020-02-28T21:00:00.000Z",
        "id": 2,
        "start": "2020-02-28T19:00:00.000Z",
      },
    ]
  }
  initialState={
    Object {
      "sortBy": Array [
        Object {
          "desc": true,
          "id": "start",
        },
      ],
    }
  }
/>
`;
